pkg/operation/botanist/component/etcd/waiter.go

```diff
@@ -20,72 +20,72 @@ import (
 	"time"
 
 	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/extensions"
 	"github.com/gardener/gardener/pkg/utils"
 	"github.com/gardener/gardener/pkg/utils/retry"
 
 	druidv1alpha1 "github.com/gardener/etcd-druid/api/v1alpha1"
-	"github.com/hashicorp/go-multierror"
-	"github.com/sirupsen/logrus"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func (e *etcd) Wait(_ context.Context) error        { return nil }
-func (e *etcd) WaitCleanup(_ context.Context) error { return nil }
+const (
+	// DefaultInterval is the default interval for retry operations.
+	DefaultInterval = 5 * time.Second
+	// DefaultSevereThreshold is the default threshold until an error reported by another component is treated as 'severe'.
+	DefaultSevereThreshold = 3 * time.Minute
+	// DefaultTimeout is the default timeout and defines how long Gardener should wait
+	// for a successful reconciliation of an Etcd resource.
+	DefaultTimeout = 5 * time.Minute
+)
 
-// WaitUntilEtcdsReady waits until all etcds in the given namespace are ready.
-func WaitUntilEtcdsReady(
-	ctx context.Context,
-	c client.Client,
-	logger logrus.FieldLogger,
-	namespace string,
-	count int,
-	interval time.Duration,
-	severeThreshold time.Duration,
-	timeout time.Duration,
-) error {
-	var retryCountUntilSevere int
+func (e *etcd) Wait(ctx context.Context) error {
+	return extensions.WaitUntilObjectReadyWithHealthFunction(
+		ctx,
+		e.client,
+		e.logger,
+		CheckEtcdObject,
+		e.etcd,
+		"Etcd",
+		DefaultInterval,
+		DefaultSevereThreshold,
+		DefaultTimeout,
+		nil,
+	)
+}
+
+func (e *etcd) WaitCleanup(_ context.Context) error { return nil }
 
-	return retry.UntilTimeout(ctx, interval, timeout, func(ctx context.Context) (done bool, err error) {
-		retryCountUntilSevere++
+// CheckEtcdObject checks if the given Etcd object was reconciled successfully.
+func CheckEtcdObject(obj client.Object) error {
```

Comment:
Should we move this to the other `Check*` functions in https://github.com/gardener/gardener/blob/master/pkg/utils/kubernetes/health/health.go?

Reply:
I explicitly avoided to add it to `pkg/utils/kubernetes/health` to not add extra dependencies for `druidv1alpha1` to the `health` package (same as for the `CheckDNSObject` func), as I thought, this package would already bring more than enough dependencies. Now that I checked again, I see that there is even already a `CheckEtcd` func with a different purpose there... 🤷‍♂️ 

Still, I think, these funcs serve a specific purpose: being used in conjunction with `WaitUntilObjectReadyWithHealthFunction` in the components' `Wait` funcs. I don't think there's currently a good use for those funcs somewhere else, so I think it makes sense to keep them in the same package where there are used. 
WDYT?

Reply:
OK, thanks, makes sense!